Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -228,6 +229,8 @@ public class ChatMessageController: DataController, DelegateCallable, DataStoreP
         do {
             try messageObserver.startObserving()
             try repliesObserver?.startObserving()
+            // TODO: Not good, this impacts the reaction query, since the initial results won't be filtered with the given query,
+            // because of this is most likely better to just create a new `ReactionListController` so that we don't break the existing message controller.
             reactions = Array(messageObserver.item?.latestReactions.sorted(by: { $0.updatedAt > $1.updatedAt }) ?? [])
```

Comment:
This is not good. The initial data, comes from latestReactions, and at this point, we don't have a reaction filter as well. In order to not break things, I prefer to create a new `ReactionListController` to make sure the new endpoint is totally isolated, and also easier to maintain than this one.

Reply:
It would be confusing for customers - on one hand you have some reactions being loaded with a message controller, on another with a new one.

Reply:
The `ChatMessageController` was designed with Message Actions UI, which also includes the reactions. But the reactions should have never been part of the message controller in the first place IMO. In Slack like UIs, you have lists of reactions. Using a `ChatMessageController` just for displaying reactions is probably even more confusing. 

If we want to make this work on `ChatMessageController` than it will require us to pass the reactions query in the initialised of the message controller in order to properly implement this, which is even more confusing for customers. Where having a `ReactionListController` is already a common pattern in the SDK. We can even deprecated the `loadReactions` from `ChatMessageController` and mention that there is this new `ReactionListController`. Or keep both, but mention that there is a new one.

Reply:
Another 2 points why using `MessageController` is not a good idea:
1. It does not follow the common pattern in the SDK, usually whenever we have a query endpoint, we have a controller.
2. When calling `MessageController.synchronize()`, it will make a request to fetch the message and start the message observers. This is actually why we set the initial reactions in the observers, and not in the `synchronize()`. We use the message controller in the `ReactionsView` without calling synchronize to avoid the additional fetch. So it is actually counter-intuitive at the moment because we always tell customers to call sync in order to get updates, for when using reactions in the Message Controller, we do not call it.